plugin/parser/differ/mysql/differ.go

```diff
@@ -237,13 +242,61 @@ func (*SchemaDiffer) SchemaDiff(oldStmt, newStmt string) (string, error) {
 					Specs: alterTableInplaceAddConstraintSpecs,
 				})
 			}
-		default:
+		case *ast.CreateViewStmt:
+			newViewList = append(newViewList, newStmt)
+		}
+	}
+
+	var tempViewList []*ast.CreateViewStmt
+	var needRealViewList []*ast.CreateViewStmt
+	for _, needView := range newViewList {
+		viewName := needView.ViewName.Name.O
+		if newNode, ok := newViewMap[viewName]; ok {
+			if !isViewEqual(needView, newNode) {
+				// skip predifined view(like temporary view in mysqldump).
+				continue
+			}
 		}
+		oldNode, ok := oldViewMap[viewName]
+		if ok {
+			if !isViewEqual(needView, oldNode) {
+				createViewStmt := needView
```

Comment:
Do we need to make a copy of the view because that's a pointer.

Reply:
We should not because we only want to modify the `OR REPLACE` field, and this field is not a pointer.
Making a copy of the `node` is a hard work with tidb parser.